httpservice/app/app.py

```diff
@@ -42,20 +44,46 @@ def create_tarfile(directory, tar_filename):
 
 @app.post("/changemaps/", dependencies=[fastapi.Security(check_apikey_header)])
 async def make_changemaps(
-    feacoll: Any = fastapi.Body(), output_tar=fastapi.Depends(sendable_tempfile)
+    data: dict = fastapi.Body(...),
+    output_tar=fastapi.Depends(sendable_tempfile)
 ):
+    input_geojson = data.get("input_geojson")
+    input_t0 = data.get("input_t0")
+    input_t1 = data.get("input_t1")
+
     with tempfile.NamedTemporaryFile(
         "w+", prefix="input-", suffix=".json"
     ) as input_fp, tempfile.TemporaryDirectory() as outdir:
-        json.dump(feacoll, input_fp)
+
+        # Dump feacoll to temp file
+        json.dump(input_geojson, input_fp)
         input_fp.flush()
         input_fp.seek(0)
 
-        gccd.flow(input_fp.name, outdir, "output")
+        # Decode and save input_t0 and input_t1 (if provided)
+        if input_t0:
+            input_t0 = save_base64_to_tempfile(input_t0_base64, suffix=".tiff")
+
+        if input_t1:
+            input_t1 = save_base64_to_tempfile(input_t1_base64, suffix=".tiff")
+
+        # Run the GCCD flow()
+        gccd.flow(input_fp.name, input_t0, input_t1, outdir, "output")
 
         create_tarfile(outdir, output_tar)
 
     return fastapi.responses.FileResponse(
         output_tar,
         headers={"Content-Disposition": "attachment; filename=changemap.tar"},
     )
+
+def save_base64_to_tempfile(base64_str, suffix=""):
+    """
+    Decode base64 string and write to a temporary file.
+    Return the path of the temporary file.
+    """
+    binary_data = base64.b64decode(base64_str)
+    temp_file = tempfile.NamedTemporaryFile(delete=False, suffix=suffix)
```

Comment:
You need a way to delete the tempfile at some point later.

I wonder if you could make this function return a bona-fide context manager that cleans the file when exiting.  ChatGPT gives us this starting point:

```py
input_t0 = "SGVsbG8gd29ybGQh"  # Example base64-encoded string

if input_t0:
    with WriteToTempFile(input_t0) as input_t0_filename:
        gccd.flow(input_fp.name, input_t0_filename, outdir, "output")(tmpf)


class WriteToTempFile:
    def __init__(self, base64_str, suffix=""):
        self.base64_str = base64_str
        self.suffix = suffix

    def __enter__(self):
        # Create a temporary file with the specified suffix
        self.temp_file = tempfile.NamedTemporaryFile(suffix=self.suffix, delete=False)

        # Decode the base64 string and write to the temporary file
        decoded_data = base64.b64decode(self.base64_str)
        self.temp_file.write(decoded_data)
        self.temp_file.close()

        # Return the temporary file name
        return self.temp_file.name

    def __exit__(self, exc_type, exc_value, traceback):
        # Delete the temporary file when the context exits
        os.remove(self.temp_file.name)

```

Reply:
If you do this, and if you need some extra cleverness to handle the case when images are not provided, you can use something like contextlib.nullcontext(), as done here:

https://github.com/ConservationMetrics/cmi-jobs/blob/9bc51a53d184a2cbb8d53d7f573f45cd6983c023/azure-functions/launch_tasks/launch_tasks_azfn.py#L460-L463

I don't know whether you will need that.

Reply:
Just looping back to say that this pretty much works fine!